Source/WebCore/page/EventHandler.cpp

```diff
@@ -3004,15 +3004,17 @@ void EventHandler::updateMouseEventTargetNode(const AtomString& eventType, Node*
             Vector<Ref<Element>, 32> leftElementsChain;
             for (Element* element = m_lastElementUnderMouse.get(); element; element = element->parentElementInComposedTree())
                 leftElementsChain.append(*element);
-            Vector<Ref<Element>, 32> enteredElementsChain;
+            Vector<WeakPtr<Element, WeakPtrImplWithEventTargetData>, 32> elementsUnderMouse;
             for (Element* element = m_elementUnderMouse.get(); element; element = element->parentElementInComposedTree())
-                enteredElementsChain.append(*element);
+                elementsUnderMouse.append(element);
 
-            if (!leftElementsChain.isEmpty() && !enteredElementsChain.isEmpty() && leftElementsChain.last().ptr() == enteredElementsChain.last().ptr()) {
+            Vector<WeakPtr<Element, WeakPtrImplWithEventTargetData>, 32> enteredElementsChain = elementsUnderMouse;
```

Comment:
I think you can drop the template args here
```suggestion
            Vector enteredElementsChain = elementsUnderMouse;
```